pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java

```diff
@@ -129,4 +129,19 @@ public WorkerInfo getClusterLeader() {
     public void rebalance() {
         worker.rebalance(uri.getRequestUri(), clientAppId());
     }
+
+    @GET
+    @ApiOperation(
+            value = "Checks if this node is the leader and is ready to service requests",
+            response = String.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "The requester is not authenticated"),
+            @ApiResponse(code = 503, message = "Worker service is not running")
+    })
+    @Path("/cluster/leader/ready")
+    @Produces(MediaType.APPLICATION_JSON)
+    public String isLeaderReady() {
```

Comment:
I think it is more appropriate to just return 200 for OK or 503 (service not available).  That is more idiomatic for HTTP REST calls

Reply:
Something like
https://github.com/apache/pulsar/blob/master/pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/WorkerReadinessResource.java#L65

Reply:
Changed